source/common/router/rds_impl.cc

```diff
@@ -206,10 +206,10 @@ void RouteConfigProviderManagerImpl::addRouteInfo(const RdsRouteConfigProvider&
                                                   Buffer::Instance& response) {
```

Comment:
can we check if response is empty and if not prepend a comma? seems simpler and encapsulates the output logic better

Reply:
agreed that seems a lot simpler. Ultimately we will swap this all out for protos with auto JSON conversion so I would just do the simple thing for now.

Reply:
the buffer won't be empty on the first addition as it will already have `[\n`. As far as I can see checking for that will be messier, unless we add a bufferToString method like in the test utils. I have consolidated in to 1 helper function to reduce complexity. 

thoughts?